### ISSUE TITLE
Flutter support Vulkan Engine on Android

### ISSUE THREAD

**Issue by tongweizhou** [flutter/flutter#93304]
@chinmaygarde @arbreng  @darshankawar 

What are some of the specific problems with Vulkan on Android? The vulkan version of the 王者荣耀（Arena Of Valor）game also runs well on Android. SurfaceFlinger is also available in the Vulkan implementation on Android.
I want to improve Flutter performance using vulkan engine, if this feature is supported it will be an overall improvement.

---------------------------------------------
https://github.com/flutter/flutter/issues/93112

**Comment by chinmaygarde** [comment]
Flutter cannot depend on the availability of Vulkan on all versions of Android it purports to support. In the absence of this guarantee, two versions of the rendering backend will have to authored, maintained, and, shipped with each version of the Flutter engine on Android. This effort will have to mirrored on the side of plugins that use the texture interop APIs as well.

This has led to the current focus on making the OpenGL ES backend for Android the preferred backend. There is nothing inherently problematic with Vulkan and other embedders use it in production. But Vulkan WSI on Android is not a team priority at the moment.

**Comment by tongweizhou** [comment]
Thanks for your patient answer. I will try to fix some bugs of vulkan , and I also expect you to give me some advice.

**Comment by iapicca** [comment]
@chinmaygarde 
imho @tongweizhou has a point, wouldn't be possible to support both Vulkan and  OpenGL ES
if there would be support for Vulkan on linux and windows some resources could be shared

**Comment by chinmaygarde** [comment]
> wouldn't be possible to support both Vulkan and OpenGL ES?

Totally. And a lot of resources are already shared. Also, we already do this on iOS where both the Metal and OpenGL backends are present in the engine (though one is heading towards deprecation). So we already know the steps involved in making such a change.

Personally, I wish we could use Vulkan everywhere as that's a modern API with many compelling advantages. I think the hesitation is in having to maintain and support (bug fixes and such) both backends for the foreseeable future. The upside of deploying these resources is perhaps not fully explored and @zanderso or @csells might have thoughts on prioritization.

**Comment by zanderso** [comment]
> Flutter cannot depend on the availability of Vulkan on all versions of Android it purports to support.

@chinmaygarde Do you know what the version requirements are?

**Comment by chinmaygarde** [comment]
No. I don't. I looked over the documentation over on Android website and it seems to indicate API 24 or above and even then Vulkan support is not guaranteed. So, unlike iOS, I don't think we can ever get to a world where we can ship just the one Vulkan backend.

**Comment by tongweizhou** [comment]
@chinmaygarde  
Vulkan always depends on the graphics driver vendor. But with ANGLE after Android Q ,the Vulkan driver is not a must ,as it can convert gles drivers to vulkan and Direct X so on.

**Comment by zanderso** [comment]
The way we'll have to make this choice over time is by weighing the benefits to performance on a wide range of devices against the increase in binary size and the additional complexity in our build and test infrastructure. Since it sounds like a Vulkan backend would benefit only the newest devices and would be pure cost with no benefit on most older (and some newer) devices, I think on balance we're not quite ready to accept changes in this direction yet. We need to keep an eye on this for opportunities, though, so thanks for bringing this up again!

**Comment by zanderso** [comment]
From feature request triage: To clarify, support for Vulkan on Android is currently best-effort. We'd be happy to take patches, but the team is not currently working on supporting it.

**Comment by gnprice** [comment]
It looks like work on this is now happening: #105644.

I think this issue can be closed as a duplicate in favor of that one.
